contrib/babelfishpg_tds/src/backend/tds/tdsbulkload.c

```diff
@@ -74,11 +74,11 @@ do \
 do \
 { \
 	if ((*message)->len - offset < dataLen) \
-		FetchMoreBcpData(message); \
+		FetchMoreBcpData(message, dataLen); \
 } while(0)
 
 static void
-FetchMoreBcpData(StringInfo *message)
+FetchMoreBcpData(StringInfo *message, int dataLenToRead)
```

Comment:
how does this help? caller was already doing this with (*message)->len - offset < dataLen check?

Reply:
The caller is only checking if it needs more data or not (If it needs more data then it callsFetchMoreBcpData ).

Inside FetchMoreBcpData  function we MIGTH need read multiple packets(case where data lies across more than 1 packet) in that case, we need to read in a while loop until dataLenToRead <= temp->len